azure/services/bastionhosts/spec.go

```diff
@@ -40,69 +41,64 @@ type AzureBastionSpec struct {
 	EnableTunneling bool
 }
 
-// AzureBastionSpecInput defines the required inputs to construct an azure bastion spec.
-type AzureBastionSpecInput struct {
-	SubnetName   string
-	PublicIPName string
-	VNetName     string
-}
-
-// ResourceName returns the name of the bastion host.
-func (s *AzureBastionSpec) ResourceName() string {
-	return s.Name
-}
-
-// ResourceGroupName returns the name of the resource group.
-func (s *AzureBastionSpec) ResourceGroupName() string {
-	return s.ResourceGroup
-}
-
-// OwnerResourceName is a no-op for bastion hosts.
-func (s *AzureBastionSpec) OwnerResourceName() string {
-	return ""
+// ResourceRef implements azure.ASOResourceSpecGetter.
+func (s *AzureBastionSpec) ResourceRef() *asonetworkv1.BastionHost {
+	return &asonetworkv1.BastionHost{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.Name,
+			Namespace: s.Namespace,
+		},
+	}
 }
 
-// Parameters returns the parameters for the bastion host.
-func (s *AzureBastionSpec) Parameters(ctx context.Context, existing interface{}) (parameters interface{}, err error) {
-	if existing != nil {
-		if _, ok := existing.(armnetwork.BastionHost); !ok {
-			return nil, errors.Errorf("%T is not an armnetwork.BastionHost", existing)
-		}
-		// bastion host already exists
-		return nil, nil
+// Parameters implements azure.ASOResourceSpecGetter.
+func (s *AzureBastionSpec) Parameters(ctx context.Context, existingBastionHost *asonetworkv1.BastionHost) (parameters *asonetworkv1.BastionHost, err error) {
+	if existingBastionHost != nil {
+		return existingBastionHost, nil
 	}
 
-	bastionHostIPConfigName := fmt.Sprintf("%s-%s", s.Name, "bastionIP")
+	bastionHost := &asonetworkv1.BastionHost{}
+	bastionHost.Spec = asonetworkv1.BastionHost_Spec{}
 
-	return armnetwork.BastionHost{
-		Name:     ptr.To(s.Name),
-		Location: ptr.To(s.Location),
-		Tags: converters.TagsToMap(infrav1.Build(infrav1.BuildParams{
-			ClusterName: s.ClusterName,
-			Lifecycle:   infrav1.ResourceLifecycleOwned,
-			Name:        ptr.To(s.Name),
-			Role:        ptr.To("Bastion"),
-		})),
-		SKU: &armnetwork.SKU{
-			Name: ptr.To(armnetwork.BastionHostSKUName(s.Sku)),
-		},
-		Properties: &armnetwork.BastionHostPropertiesFormat{
-			EnableTunneling: ptr.To(s.EnableTunneling),
-			DNSName:         ptr.To(fmt.Sprintf("%s-bastion", strings.ToLower(s.Name))),
-			IPConfigurations: []*armnetwork.BastionHostIPConfiguration{
-				{
-					Name: ptr.To(bastionHostIPConfigName),
-					Properties: &armnetwork.BastionHostIPConfigurationPropertiesFormat{
-						Subnet: &armnetwork.SubResource{
-							ID: &s.SubnetID,
-						},
-						PublicIPAddress: &armnetwork.SubResource{
-							ID: &s.PublicIPID,
-						},
-						PrivateIPAllocationMethod: ptr.To(armnetwork.IPAllocationMethodDynamic),
-					},
+	bastionHostIPConfigName := fmt.Sprintf("%s-%s", s.Name, "bastionIP")
+	bastionHost.Spec.AzureName = s.Name
+	bastionHost.Spec.Location = ptr.To(s.Location)
```

Comment:
I am a little divided in here. Should this be 
```suggestion
	if s.Location != "" {
		bastionHost.Spec.Location = ptr.To(s.Location)
	}
```

In this way, the field is only set when `s.Location` is not empty.?